### DISCUSSION TITLE
Waverly: a community platform built on Bluesky

### DISCUSSION THREAD

**Discussion by PhilBeaudoin** [bluesky-social/atproto#2306] [Project Show and Tell]
We're getting ready to unveil [Waverly](https://waverly.social): a new social platform for communities. Think Facebook group, but that really cares about nurturing your community and helping it grow in the direction you want...

...and the first step towards nurturing a good community is building it on a network that doesn't lock you in! That's why we built Waverly on Bluesky. AtProto, PDS, AppView: we eat that stuff every day. :)

If you're part of a community that would like this kind of stuff, please reach out. If you're a dev curious about how we did this, I'm on the [Bluesky Discord channel](https://discord.com/channels/1207024379549061120/1207050328063610923) every day, come say hi.

![image](https://github.com/bluesky-social/atproto/assets/229691/63500704-3a17-4740-987f-2e9e79b85b2b)

**Comment by DavidBuchanan314** [comment]
I'm curious to know to what extent Waverly integrates with the existing atproto/Bluesky ecosystem, vs being a separate entity built on top of the same core tech. (And I don't mean to imply that one approach might be better than the other!)

Can regular Bluesky users (on bsky.app) see or interact with Waverly content?

Did you need to extend or create new lexicon definitions to make this work?

Likewise, did you need to extend the pds, relay, or appview components?

**Comment by PhilBeaudoin** [comment]
Short answer: It's built with federation in mind. We want to eventually live on the network and interop with many other social networks.

Long answer:
Our identity is fully compatible. We use standard PDS lexicon for all that stuff.

For the app view, we use the bsky lexicon as much as possible. For example, our profile is the same as the bsky profile. We had to create our own post format because we needed to increase the character limit, but it's the only difference in the schema. Likes, follows, replies, embeds... for all of that we use bsky schema. As a result it would be relatively trivial for the bsky app view to ingest our posts. (We have some extra schemas, and some of the bsky stuff we don't render well in the app.)

Our communities (we call them _waves_ 😊)  are just actors that can be controlled by multiple people. We keep the actor's credentials on a private server and do some very basic ACL so that the community admins and community members can control the bots. A post to the community is just a post by a user reshared by the wave-actor.

We built our own AppView, but it's strongly inspired by the bsky app view. (We needed to adjust notifications to account for our wave actors, etc.)

Our PDSs are not federated yet and we cannot ingest from external PDSs yet. I think both us and the federation have to "grow in maturity" a bit before we take it to the next level. ;) However, our hope is to get there.

**Comment by erlend-sh** [comment]
Cool! Very excited to see how this will interoperate with the likes of Bluesky and others.

> Waverly is built around the idea that social media can be built on a foundation of transparency and user control, with the goal of empowering users rather than manipulating them.

I consider source code availability an essential part of transparency and user control. Is this planned?

**Reply by PhilBeaudoin** [reply]
There are definitely part of the code we want to open source. This will come later, however, as we’re operating on a shoestring budget and can’t find time to untangle the code we want to open source.

That being said, we are a VC-backed company* and need to find ways to bring returns. We do not want to do it by keeping users captive in our ecosystem, so we’ll need to find it elsewhere. Will we need to keep some of the code closed? Or go for some open core approach? We’re not sure yet… First order of business is to put something out there and see if people pick it up.

* I know some of you feel strongly about the  mere concept of a VC-backed software company. I totally respect that and thought a long time before deciding to go down this road with Waverly. It is definitely a trade-off.

**Reply by mikestaub** [reply]
There is nothing wrong with taking investor money as long as you are aware of the risks it poses to the architecture early on. Jay Graber was prudent enough to form a PB LLC before raising funds, to align the Bylaws with the mission. OpenAI was almost destroyed by not thinking enough about governance early on. I'm sure there will be many closed source tools that interact with ATproto, what is important in my opinion is that they don't inject proprietary data schemas into the PDS records. A great example would be if someone spends a lot of money to fine-tune an LLM to be amazing at labeling posts in a specific way with 99.9% accuracy and wants to charge for this service. Then of course, they would not want to open source everything. A more hostile example would be an AppView implementing a custom DM feature that uses closed source proprietary encryption.

**Reply by PhilBeaudoin** [reply]
@mikestaub Thanks for these thoughtful words. Structure-wise, we're just a very regular (Canadian) inc. Value- and mission-wise we're aligned with helping make the open ecosystem better, and we're very vocal about that. Our early investors know and understand this — these includes Mozilla Ventures, btw, and I believe our value-alignment played an important part in their decision to invest. That being said, I'm not naive and I know startups have a life of their own and that sometimes the best intentions can be thwarted by future investors. The only thing I can say, right now, is that I will personally hold strong to these values.

**Comment by thomasgwatson** [comment]
Is waverly still running? Website doesn't seem to work

**Comment by cartcodes** [comment]
Interested, but the website is down.

**Comment by RonDayVous** [comment]
I'm with the others, The website is down, is this a bust?

**Comment by PhilBeaudoin** [comment]
Hello! Yes, we unfortunately had to take the website down and move to other projects for lack of funding. Happy to share our findings.

**Reply by erlend-sh** [reply]
Sorry to hear that; would love to hear about your findings 🙏

**Comment by RonDayVous** [comment]
I just received 2 invoices, 1 from you and another from Alina Fischer. The problem is that neither reflects my request to be upgraded to the 

Non-Profit 500 plan.


I have sent several emails about this and am a little confused about why this is taking so long.


Thanks


Ron Day
115 Mott Street
Milford, PA 18337
917-544-2976

On Monday, March 10, 2025 19:04:31 (-04:00), Philippe Beaudoin wrote:



Hello! Yes, we unfortunately had to take the website down and move to other projects for lack of funding. Happy to share our findings.
